src/dune/context.ml

```diff
@@ -551,27 +551,40 @@ let default ~merlin ~env_nodes ~env ~targets ~fdo_target_exe
     ~disable_dynamically_linked_foreign_archives
 
 let opam_version =
-  let res = ref None in
-  fun opam env ->
-    match !res with
-    | Some future -> Fiber.Future.wait future
-    | None ->
-      let* future =
-        Fiber.fork (fun () ->
-            let+ version =
-              Process.run_capture_line Strict ~env opam [ "--version" ]
-            in
-            match Scanf.sscanf version "%d.%d.%d" (fun a b c -> (a, b, c)) with
-            | Ok s -> s
-            | Error () ->
-              User_error.raise
-                [ Pp.textf "`%s config --version' returned invalid output:"
-                    (Path.to_string_maybe_quoted opam)
-                ; Pp.verbatim version
-                ])
-      in
-      res := Some future;
-      Fiber.Future.wait future
+  let module Input = struct
+      type t = Path.t * Env.t
+
+      let to_dyn = Tuple.T2.to_dyn Path.to_dyn Env.to_dyn
+
+      let equal (p1, e1) (p2, e2) = Path.equal p1 p2 && Env.equal e1 e2
+
+      let hash = Tuple.T2.hash Path.hash Env.hash
+    end
+  in
+  let f (opam, env) =
+    let+ version =
+      Process.run_capture_line Strict ~env opam [ "--version" ]
```

Comment:
IIRC, the only reason we pass the environment explicitly here is for `OPAMCOLOR`. What about removing the setting of `OPAMCOLOR` and passing `--color always` instead? cf #2339

That would simplify this code.

Reply:
Sure. Although in this case, we don't even need this option then. I don't think opam ever prints the version with color anyways.

PS: do you mean `--color never`?

Reply:
> Sure. Although in this case, we don't even need this option then. I don't think opam ever prints the version with color anyways.

That's very true.

> PS: do you mean --color never?

I did mean `--color always`, because `dune` parses ansi escape sequences. However, it is true that in this case we don't want colors since we are capturing the output.